src/services/severa-api-service.ts

```diff
@@ -421,7 +426,213 @@ export const CreateSeveraApiService = (): SeveraApiService => {
         throw new Error(`Failed to fetch work hours: ${response.status} - ${response.statusText}`);
       }
       return response.json();
-    }
+    },
+
+    /**
+     * Gets resource allocations for a specific user or for all opt-in users if no user is specified.
+     *
+     * @param severaUserId - The GUID of the Severa user (optional).
+     * @param optInUsers - Array of opted-in users ({ guid: string }[]).
+     * @returns {Promise<SeveraResponseResourceAllocation[]>} - Array of resource allocations.
+     */
+    getResourceAllocationsByUserOrAll: async (
+      severaUserId: string | undefined,
+      optInUsers: { guid: string }[]
+    ): Promise<SeveraResponseResourceAllocation[]> => {
+      const buildResourceAllocationUrl = (severaUserId?: string): URL | null => {
+        let endpointPath: string;
+        if (severaUserId) {
+          endpointPath = `users/${severaUserId}/resourceallocations/allocations`;
+          const customUrl = new URL(`${baseUrl}/v1/${endpointPath}`);
+          return customUrl;
+        } else {
+          return null;
+        }
+      };
+
+      const url = buildResourceAllocationUrl(severaUserId);
+      let allResourceAllocations: SeveraResponseResourceAllocation[] = [];
+
+      if (url) {
+        const response = await (await fetch(url.toString(), {
+          method: "GET",
+          headers: {
+            Authorization: `Bearer ${await getSeveraAccessToken()}`,
+            Client_Id: process.env.SEVERA_DEMO_CLIENT_ID,
```

Comment:
Could this also be a good PR to refactor the SEVERA_DEMO_CLIENT_ID env, we should add a SEVERA_CLIENT_ID env, both should be optional (and exclusive) so one must be present, both will not be.
Locally we will only be using the demo client but for the production we will add the production id (and not the demo id)

Reply:
This update should be done everywheere the demo client is currently used to make sure it will work when we switch to production environemnt